src/merkle/mod.rs

```diff
@@ -42,6 +45,7 @@ pub enum MerkleError {
     InvalidPath(MerklePath),
     InvalidEntriesCount(usize, usize),
     NodeNotInSet(u64),
+    NodeNotInStorage(Word, NodeIndex),
```

Comment:
That is used in the `MerklePathSet` structure, which is a partial view, so I interpreted it to mean "this would have been here if we had the full view, but this partial view is missing the value", i wasn't sure how much of an overlap there was, so I added another error. Let me know if they have to be merged.

Reply:
Yes, I think these are different errors - but the naming is too close. Maybe name this one `NodeNoteFound`?

Reply:
I honestly think it is a bit confusing to differentiate one from the other, but we can introduce the new variant now and disambiguate later